packages/http-server/src/getHttpConfigFromRequest.ts

```diff
@@ -41,7 +45,7 @@ export const getHttpConfigFromRequest: PrismConfigFactory<IHttpConfig, IHttpRequ
     if (typeof config.mock === 'boolean') {
       config.mock = httpOperationConfig;
     } else {
-      config.mock = Object.assign(config.mock, httpOperationConfig);
+      config.mock = Object.assign({}, config.mock, httpOperationConfig);
```

Comment:
:) @XVincentX removed this ~week ago

Reply:
I will leave it you to you guys ;)

Reply:
Since `config.mock` may be undefined (which is basically concept of this PR) the above is needed